Golang 1.17 broke Tests with crypto/x509

CreateCertificate now returns an error if the provided private key doesn't match the parent's public key.
https://go-review.googlesource.com/c/go/+/224157/

**Test Logs**
```
?       github.com/square/certstrap     [no test files]                                                                                                                                                        
=== RUN   TestParseExpiryWithSeconds                                                                                                                                                                           
--- PASS: TestParseExpiryWithSeconds (0.00s)                                                                                                                                                                   
=== RUN   TestParseExpiryWithMinutes                                                                                                                                                                           
--- PASS: TestParseExpiryWithMinutes (0.00s)                                                                                                                                                                   
=== RUN   TestParseExpiryWithHours                                                                                                                                                                             
--- PASS: TestParseExpiryWithHours (0.00s)                                                                                                                                                                     
=== RUN   TestParseExpiryWithDays                                                                                                                                                                              
--- PASS: TestParseExpiryWithDays (0.00s)                                                                                                                                                                      
=== RUN   TestParseExpiryWithMonths                                                                                                                                                                            
--- PASS: TestParseExpiryWithMonths (0.00s)                                                                                                                                                                    
=== RUN   TestParseExpiryWithYears                                                                                                                                                                             
--- PASS: TestParseExpiryWithYears (0.00s)                                                                                                                                                                     
=== RUN   TestParseExpiryWithMixed                                                                                                                                                                             
--- PASS: TestParseExpiryWithMixed (0.00s)                                                                                                                                                                     
=== RUN   TestParseInvalidExpiry                                                                                                                                                                               
--- PASS: TestParseInvalidExpiry (0.00s)                                                                                                                                                                       
=== RUN   TestRevokeCmd                                                                                                                                                                                        
    revoke_test.go:118: could not create cert host: x509: provided PrivateKey doesn't match parent's PublicKey                                                                                                 
--- FAIL: TestRevokeCmd (0.12s)                                                                                                                                                                                
FAIL                                                                                                                                                                                                           
FAIL    github.com/square/certstrap/cmd 0.122s                                                                                                                                                                 
=== RUN   TestDepotCRUD                                                                                                                                                                                        
--- PASS: TestDepotCRUD (0.01s)                                                                                                                                                                                
=== RUN   TestDepotPutNil                                                                                                                                                                                      
--- PASS: TestDepotPutNil (0.01s)                                                                                                                                                                              
=== RUN   TestDepotCheckFailure                                                                                                                                                                                
--- PASS: TestDepotCheckFailure (0.01s)                                                                                                                                                                        
=== RUN   TestDepotGetFailure                                                                                                                                                                                  
--- PASS: TestDepotGetFailure (0.01s)                                                                                                                                                                          
=== RUN   TestDepotList                                                                                                                                                                                        
--- PASS: TestDepotList (0.01s)                                                                                                                                                                                
=== RUN   TestDepotGetFile
--- PASS: TestDepotGetFile (0.01s)
PASS                                               
ok      github.com/square/certstrap/depot       0.047s
=== RUN   TestCreateCertificateAuthority
--- PASS: TestCreateCertificateAuthority (0.01s)
=== RUN   TestCreateCertificateHost
--- PASS: TestCreateCertificateHost (0.00s)
=== RUN   TestCertificateAuthorityInfo
--- PASS: TestCertificateAuthorityInfo (0.00s)
=== RUN   TestCertificateAuthorityInfoFromJSON
--- PASS: TestCertificateAuthorityInfoFromJSON (0.00s)
=== RUN   TestCertificateAuthority
--- PASS: TestCertificateAuthority (0.00s)
=== RUN   TestWrongCertificate
--- PASS: TestWrongCertificate (0.00s)
=== RUN   TestBadCertificate
--- PASS: TestBadCertificate (0.00s)
=== RUN   TestCertificateVerify
--- PASS: TestCertificateVerify (0.00s)
=== RUN   TestCreateCertificateRevocationList                                                                                                                                                        [299/1166]
--- PASS: TestCreateCertificateRevocationList (0.02s)
=== RUN   TestCertificateRevocationList
--- PASS: TestCertificateRevocationList (0.00s)
=== RUN   TestCreateCertificateSigningRequest
--- PASS: TestCreateCertificateSigningRequest (0.03s)
=== RUN   TestCertificateSigningRequest
--- PASS: TestCertificateSigningRequest (0.00s)
=== RUN   TestOldStyleCertificateSigningRequest
--- PASS: TestOldStyleCertificateSigningRequest (0.00s)
=== RUN   TestWrongCertificateSigningRequest
--- PASS: TestWrongCertificateSigningRequest (0.00s)
=== RUN   TestBadCertificateSigningRequest
--- PASS: TestBadCertificateSigningRequest (0.00s)
=== RUN   TestCreateRSAKey
--- PASS: TestCreateRSAKey (0.03s)
=== RUN   TestRSAKey                               
--- PASS: TestRSAKey (0.00s)
=== RUN   TestWrongRSAKey
--- PASS: TestWrongRSAKey (0.00s)
=== RUN   TestBadRSAKey                            
--- PASS: TestBadRSAKey (0.00s)
=== RUN   TestRSAKeyExport
--- PASS: TestRSAKeyExport (0.00s)
=== RUN   TestRSAKeyExportEncrypted
--- PASS: TestRSAKeyExportEncrypted (0.00s)
=== RUN   TestRSAKeyGenerateSubjectKeyID
--- PASS: TestRSAKeyGenerateSubjectKeyID (0.00s)
PASS                                               
ok      github.com/square/certstrap/pkix        0.092s
=== RUN   TestVersion                          
--- PASS: TestVersion (0.08s)
=== RUN   TestIp                                   
--- PASS: TestIp (0.26s)                           
=== RUN   TestNotCA                                
--- PASS: TestNotCA (2.07s)
=== RUN   TestURI                                  
--- PASS: TestURI (0.28s)
=== RUN   TestWorkflow                             
--- PASS: TestWorkflow (1.21s)
PASS                                               
ok      github.com/square/certstrap/tests       3.907s
FAIL
```
**System**
Go version: go version go1.17.1 linux/amd64
Certstrap version: 17687043c6aa40489620cfba2d6350d586cda8ed
Linux version: Arch linux

--- a/cmd/revoke_test.go
+++ b/cmd/revoke_test.go
@@ -113,7 +113,12 @@ func setupCN(t *testing.T, dt depot.Depot) {
 		t.Fatalf("could not get cert: %v", err)
 	}
 
-	cnCert, err := pkix.CreateCertificateHost(caCert, key, csr, time.Now().Add(1*time.Hour))
+	caKey, err := depot.GetPrivateKey(dt, caName)
+	if err != nil {
+		t.Fatalf("could not get CA key: %v", err)
+	}
+
+	cnCert, err := pkix.CreateCertificateHost(caCert, caKey, csr, time.Now().Add(1*time.Hour))
 	if err != nil {
 		t.Fatalf("could not create cert host: %v", err)
 	}

